scripts/build/builders/imx.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from enum import Enum, auto
+
+from .host import HostBuilder
+
+
+class IMXApp(Enum):
+    CHIP_TOOL = auto()
+    LIGHT = auto()
+
+    def ExamplePath(self):
+        if self == IMXApp.CHIP_TOOL:
+            return 'chip-tool'
+        if self == IMXApp.LIGHT:
+            return 'lighting-app/linux'
+
+    def OutputNames(self):
+        if self == IMXApp.CHIP_TOOL:
+            yield 'chip-tool'
+            yield 'chip-tool.map'
+        if self == IMXApp.LIGHT:
+            yield 'chip-lighting-app'
+            yield 'chip-lighting-app.map'
+
+
+class IMXBuilder(HostBuilder):
```

Comment:
do you need HostBuilder? Could you use GnBuilder instaed as that is meant to be generic?

Reply:
done.